### ISSUE TITLE
Maximum call stack size exceeded for self-referencing datatypes

### ISSUE THREAD

**Issue by johnchandlerburnham** [HigherOrderCO/Kind#23]
```
T Number { lte : Number -> Number -> Type}      
| cut                  
   { left : Number(lte) 
   , right : Number(lte)   
   , ord : lte(left, right)
  }
```

produces

```
RangeError: Maximum call stack size exceeded
    at Typ (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:38:47)
    at unquote (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:776:26)
    at norm (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:887:18)
    at typecheck (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:1382:30)
    at typecheck (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:1426:9)
    at typecheck (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:1666:37)
    at typecheck (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:1405:22)
    at typecheck (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:1405:22)
    at typecheck (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:1426:9)
    at typecheck (/nix/store/lmbq47v660mf4svs6j9qn62p2p897ix7-node_formality-lang-0.3.117/lib/node_modules/formality-lang/src/fm-core.js:1666:37)
```

**Comment by johnchandlerburnham** [comment]
Simpler example:

```
T Number { a : Number }
| foo
```

**Comment by VictorTaelin** [comment]
Fixed on 11cb3b1002268fdcb9e3c9840e57cfc93455a5ed
